src/components/services/background/background.js

```diff
@@ -0,0 +1,104 @@
+angular.module("bawApp.components.background", [])
+    .service("bgImage", function() {
+        var curBg = "";
+        return {
+            curBg: curBg
+        };
+    })
+    .component("background",{
+        template: "<div ng-repeat='bg in bgs' ng-if=\"bg.state !== 'finished'\" class='background {{bg.state}}' style='background-image:url({{bg.url}});'></div>",
+        controller: ["$scope", "bgImage", function ($scope, bgImage) {
+
+            var self = this;
+
+            $scope.bgs = [];
+
+            /**
+             * Watches for changes to the bgImage service. When a change is detected it
+             * adds to the bgs array on scope. The array holds the url and the 'state'. State can be
+             * preloading, fadeIn, done and finished.
+             * preloading: image is preloading. Set the opacity to 0
+             * fadeIn: image has been preloaded, fade it in.
+             * done: image is finished fading in. Anything behind it can be removed.
+             * finished: is behind an image with state done. Remove it from dom.
+             * The bgs array is bound to scope in a repeat. New images can be added quicker than they can fade in without problem
+             * because it just adds them over the top of the existing fading-in images.
+             * @param newUrl
+             * @param oldUrl
+             */
+
+            self.setBg = function () {
+
+                if (typeof bgImage.curBg !== "string" || bgImage.curBg.length === 0) {
+                    return;
```

Comment:
When the page is loading the watcher is triggered, even though both the oldval and newval are empty strings. I didn't really check why it was being triggered, I just figured it was something to do with the the order that angular loads things.  
![image](https://cloud.githubusercontent.com/assets/5986336/24643681/d9dc912c-1952-11e7-871b-fd5795b75f18.png)